### ISSUE TITLE
MultiResponse with output queue leads to: Exception: System.ObjectDisposedException: IFeatureCollection has been disposed. Object name: 'Collection'

### ISSUE THREAD

**Issue by AnthonyGiretti** [Azure/azure-functions-dotnet-worker#2787]
### Description
I'm experiencing issues with my Azure Functions while using the MultiResponse on HttpTriggers.
Once a single request it's working, once several requests are coming in, it leads to the following error: 
'Exception while executing function: Functions.IqIngressFunction Result: Failure
Exception: System.ObjectDisposedException: IFeatureCollection has been disposed.
Object name: 'Collection'.'

### Expected behavior
It should not throw that error

### Actual behavior
It's working correctly locally on my machine, the issue appears only when it's deployed on Azure.
When I remove ASP.NET Core Integration and returning only the messages (string[]) in output it's working
When I remove the output queue it's also working (returning only IActionResult)
Only when we mix Httpresponse management with ASP.NET Core Integration + output queue it's crashing on the cloud only, not locally

### Relevant source code snippets
```csharp
public class MultiResponse
{
    [QueueOutput("%SmsIngressQueueName%", Connection = Constants.StorageAccount.ConnectionString)]
    public string[] Messages { get; set; }

    [HttpResult]
    public IActionResult HttpResult { get; set; }
}

public class IqIngressFunction
{
    private readonly IIqSmsService _iqSmsService;
    private readonly ILogger<IqIngressFunction> _logger;

    public IqIngressFunction(IIqSmsService iqSmsService, ILogger<IqIngressFunction> logger)
    {
        _iqSmsService = iqSmsService;
        _logger = logger;
    }


    [Function(nameof(IqIngressFunction))]
    public async Task<MultiResponse> Run([HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = null)] HttpRequest req)
    {
        var output = new List<string>() { };
        var multiResponse = new MultiResponse();

        try
        {
            string requestBody = string.Empty;
            using (StreamReader streamReader = new StreamReader(req.Body))
            {
                requestBody = await streamReader.ReadToEndAsync();
            }

            _logger.LogInformation($"HTTP trigger function processed a request for IqIngressFunction, body: {requestBody}");

            var inbound = JsonConvert.DeserializeObject<IqInboundModel>(requestBody);
            if (inbound != null && inbound.deliveryReceipt)
            {
                multiResponse.Messages = new string[] { };
                multiResponse.HttpResult = new OkObjectResult("ok");
                return multiResponse;
            }
            var validMsg = _iqSmsService.ValidateIngressMessage(inbound);
            if (!string.IsNullOrEmpty(validMsg))
            {
                throw new Exception(validMsg);
            }

            var msg = new SmsMessage
            {
                Sender = inbound.from,
                Receiver = inbound.to[0],
                Message = inbound.text,
                Smscid = "IQ-x"
            };
            output.Add(JsonConvert.SerializeObject(msg));

            multiResponse.Messages = output.ToArray();
            multiResponse.HttpResult = new OkObjectResult("ok");
            return multiResponse;
        }
        catch (Exception ex)
        {
            _logger.LogError(ex, $"IqIngressFunction - An unexpected error occurred. Details: {ex.Message}");

            multiResponse.Messages = new string[] { };
            multiResponse.HttpResult = new BadRequestObjectResult(ex.Message);
            return multiResponse;
        }
    }
}
```

### Known workarounds
Using a queue client (SDK required works)

### App Details

```
<ItemGroup>
	<PackageReference Include="Azure.Identity" Version="1.12.0" />
  <PackageReference Include="Apache.NMS.ActiveMQ.NetStd" Version="1.8.0" />
  <PackageReference Include="Azure.Storage.Queues" Version="12.19.1" />
  <PackageReference Include="Microsoft.Azure.Cosmos" Version="3.42.0" />
	<PackageReference Include="Microsoft.Azure.Functions.Extensions" Version="1.1.0" />
	<PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore" Version="1.3.2" />
	<PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Storage.Queues" Version="5.5.0" />
	<PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.17.4" />
	<PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.23.0" />
	<PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Warmup" Version="4.0.2" />
	<PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.EventGrid" Version="3.4.2" />
	<PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.EventHubs" Version="6.3.6" />
	<PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.OpenApi" Version="1.5.1" />
	<PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Storage" Version="6.6.0" />
	<PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http" Version="3.2.0" />
	<PackageReference Include="Microsoft.Azure.Functions.Worker.ApplicationInsights" Version="1.4.0" />
	<PackageReference Include="Microsoft.Azure.AppConfiguration.Functions.Worker" Version="7.3.0" />
	<PackageReference Include="Microsoft.ApplicationInsights.WorkerService" Version="2.22.0" />
	<PackageReference Include="Microsoft.Extensions.Configuration.AzureAppConfiguration" Version="7.3.0" />
  <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
  <PackageReference Include="Twilio" Version="5.79.0" />
```
</ItemGroup>

- .NET 8 Isolated mode
- Azure function with HttpTrigger
- **Azure Functions runtime version v4**:
- **C#*:

### Screenshots
![Image](https://github.com/user-attachments/assets/d8fbdfdd-d2c2-4c0e-ab31-7cc2bbbf3c22)

### If deployed to Azure
see screenshot below

- **Timeframe issue observed:**:Happening right after the deployment happened, NOT HAPPENING ON LOCAL MACHIN

### Steps to reproduce

Using MultiResponse as follows:

```csharp
public class MultiResponse
{
    [QueueOutput("%SmsIngressQueueName%", Connection = Constants.StorageAccount.ConnectionString)]
    public string[] Messages { get; set; }

    [HttpResult]
    public IActionResult HttpResult { get; set; }
}
```

Then create an HttpClient and run 10 calls. 50 % of them will fail (InternlServerError 500), while others will work and return HttpStatus 200 OK

**Comment by fabiocav** [comment]
@satvu let's sync. This shouldn't be using the `HttpResponseData` path.

**Comment by satvu** [comment]
@AnthonyGiretti We have been unable to repro this issue with `IActionResult` - have you still seen this issue recently? Could you share a recent invocation and the app privately through [these steps](https://github.com/Azure/azure-functions-host/wiki/Sharing-Your-Function-App-name-privately)? Lastly, is there any chance the deployment payload was stale? 

This is a known issue in MultiResponse with `HttpResponseData` under the new HTTP model, but the code path in the stack trace should not be hit when `IActionResult` is used.

**Comment by AnthonyGiretti** [comment]
@satvu  Hi, the issue is still happening. But only when deployed in the cloud. It's never happening on my local machine. The issue stopped when we rollbacked to .NET6 in process. I can't right away send what you are asking, it will take me some days. 
We have removed the MultiResponse and replaced the queue attribute by the usage  of the azure storage queue sdk instead. 
No issue when the HttpTrigger only returns IActionResult, it's only happening when it's coupled to the QueueOutput attribute in a MultiResponse. I suggest you to deploy on it on Azure, make a load test, sending 15 httprequest in a loop, the first call will work and all other will fail.

I'll get back to you in a few days

**Comment by AnthonyGiretti** [comment]
@satvu Can the following help?  I have been to AI to find the errors:

Region: WestUS 2

![Image](https://github.com/user-attachments/assets/f4c9bfe8-ddfa-4d1e-aeed-b45b85d1fc87)

Operation Id | ddb04eac3c6193e217f504e8d5ac381d
Event time | 10/16/2024, 12:34:12.2047185 PM (Local time) --> Canada Eastern time
prop__EndTime | 2024-10-16T16:34:12.2040000Z
prop__StartTime | 2024-10-16T16:34:12.1640000Z
InvocationId | 91a5575b-3db2-4ea7-83eb-1133656c9d86

**Comment by satvu** [comment]
@AnthonyGiretti Thanks, we will take a look.

**Comment by satvu** [comment]
After extensive investigation, we were unable to repro this with a simple multi-output scenario which included a queue output and `IActionResult`. Load tests for it were unable to produce the same error, but we got a matching error when using `HttpResponseData` only. Internal tests that use multi-output have not reported this problem with `IActionResult` either (#2682 was discovered internally). 

Closing this and tracking the fix for this error in #2682. I
